### ISSUE TITLE
Ignoring annotations imported from __future__

### ISSUE THREAD

**Issue by microcoder** [microsoft/pyright#138]
Hi! I'm not sure this is a bug.

**Describe the bug**
When I annotate a parameter types that declared after it, then the Python raise an exception: `NameError: name 'ClassB' is not defined`. It is not a problem.
If import annotations from `__future__` the Python handles it correctly and don't raise a exception but the Pyright don't account it.

**To Reproduce**
```python
from __future__ import annotations


class ClassA():

    def __init__(self, b: ClassB):  # 'ClassB' is unbound pyright
        pass


class ClassB():
    pass


b = ClassB()
a = ClassA(b)
```

**Screenshots or Code**
![Screenshot_20190528_080252](https://user-images.githubusercontent.com/2853555/58452197-34201d80-811f-11e9-9137-9a9c88121a22.png)

**VS Code extension or command-line**
Running pyright as a VS Code extension 1.0.34

**Comment by erictraut** [comment]
Thanks for the suggestion. You are correct that pyright doesn't currently support this `__future__` feature. To work around this, you can enclose the annotation in quotes.

**Comment by erictraut** [comment]
Thanks again for the suggestion. This is now implemented and will be in the next version of pyright.

**Comment by erictraut** [comment]
This is now implemented in version 1.0.38, which I just published.
